scripts/postdeploy.sh

```diff
@@ -7,4 +7,220 @@ for cmd in "${commands[@]}"; do
+      break
+    fi
+    sleep 20
+    Resp=$(curl -X GET -H "Authorization: Bearer $AccessToken" "$TestProfileRunURL")
+    echo "Current Status: $(echo "$Resp" | jq -r '.status'), PollCount: $PollCount"
+    PollCount=$((PollCount + 1))
+  done
+}
+
+# Ensure az load extension is installed
+az extension add --name load
+
+# Get Dataplane URL for Load Testing Resource
+DataPlaneURL=$(az load show -n "$LoadTestResourceName" -g "$ResourceGroupName" --query dataPlaneURI)
+DataPlaneURL="https://$(echo "$DataPlaneURL" | tr -d '"')"
+echo "DataplaneURL: $DataPlaneURL"
+
+# Create Load Test
+echo "Creating test with testId: $TestId"
+if az load test show --name "$LoadTestResourceName" --test-id "$TestId" --resource-group "$ResourceGroupName"; then
+  echo "Test with ID: $TestId already exists"
+  az load test update --name "$LoadTestResourceName" --test-id "$TestId" --display-name "$LoadTestDisplayName" --resource-group "$ResourceGroupName" --engine-instances "$EngineInstances"
+else
+  echo "Test with ID: $TestId does not exist. Creating a new test"
+  az load test create --name "$LoadTestResourceName" --test-id "$TestId" --display-name "$LoadTestDisplayName" --resource-group "$ResourceGroupName" --engine-instances "$EngineInstances"
+fi
+echo "Successfully created a load test"
+
+# Upload Test Plan
+echo "Upload test plan to test with testId: $TestId"
+TestPlan=$(Get_UrlTestConfig)
+TestPlanUploadURL="$DataPlaneURL/tests/$TestId/files/$TestFileName?api-version=$ApiVersion&fileType=URL_TEST_CONFIG"
+
+TestPlanUploadResp=$(Upload_TestFile "$TestPlanUploadURL" "$TestPlan")
+echo "Successfully uploaded the test plan to the test"
+
+# Create Test Profile
+TestProfileRequest=$(cat <<EOF
+{
+  "displayName": "$TestProfileDisplayName",
+  "description": "$TestProfileDescription",
+  "testId": "$TestId",
+  "targetResourceId": "$FunctionAppResourceId",
+  "targetResourceConfigurations": {
+    "kind": "FunctionsFlexConsumption",
+    "configurations": {
+      "config1": {
+        "instanceMemoryMB": "2048",
+        "httpConcurrency": 1
+      },
+      "config2": {
+        "instanceMemoryMB": "2048",
+        "httpConcurrency": 4
+      },
+      "config3": {
+        "instanceMemoryMB": "2048",
+        "httpConcurrency": 16
+      },
+      "config4": {
+        "instanceMemoryMB": "4096",
+        "httpConcurrency": 1
+      },
+      "config5": {
+        "instanceMemoryMB": "4096",
+        "httpConcurrency": 4
+      }
+    }
+  }
+}
+EOF
+)
+TestProfileResp=$(Call_AzureLoadTesting "$DataPlaneURL/test-profiles/$TestProfileId?api-version=$ApiVersion" 'PATCH' "$TestProfileRequest")
+echo "Successfully created the test profile"
+
+# Create Test Profile Run
+TestProfileRunRequest=$(cat <<EOF
+{
+  "testProfileId": "$TestProfileId",
+  "displayName": "$TestProfileRunDisplayName"
+}
+EOF
+)
+TestProfileRunId=$(uuidgen)
+TestProfileRunURL="$DataPlaneURL/test-profile-runs/$TestProfileRunId?api-version=$ApiVersion"
```

Comment:
Script seems to have an issue, it didnt call Patch test profile run?

Reply:
WIP.. Coming in next iteration

Reply:
Added it in this iteration